lib/logflare/backends/adaptor/slack_adaptor.ex

```diff
@@ -59,33 +96,64 @@ defmodule Logflare.Backends.Adaptor.SlackAdaptor do
   - list of maps
   - map
 
-  ### Example
-  The text will be prefixed with a bullet point, with a colon separating the key-value pair.
-    iex> to_markdown(%{"test"=> "test"})
-    "• test: test"
+  Text will have a colon separating the key-value pair.
+  Multi-line strings will be separated with a line break
+  Keys will be sorted alphabetically with a line break in between each key
+  If link is present in the results, it will be conveted into a url
+  """
+  @spec to_rich_text_preformatted([map()] | map()) :: [map()]
 
-  Keys within the map will be placed in an indentedrow below, sorted by key alphabetically.
-  Indentation uses 4 spaces.
-    iex> to_markdown(%{a: "test", b: "test"})
-    "• a: test\r    b: test"
+  def to_rich_text_preformatted(%{} = row) do
+    row
+    |> Enum.sort_by(fn {k, _} -> k end)
+    |> Enum.map(fn {k, v} ->
+      v_str = stringify(v)
 
-  Muliple objects will be converted into multi-line strings
-    iex> to_markdown([%{one: "test"}, %{two: "test"}])
-    "• one: test\r• two: test"
+      cond do
+        is_number(v) and String.length(v_str) == 16 ->
+          # convert to timestamp
+          {:ok, dt} = DateTime.from_unix(v, :microsecond)
 
-  """
-  @spec to_markdown([map()] | map()) :: [String.t()] | String.t()
-  def to_markdown(rows) when is_list(rows), do: Enum.map(rows, &to_markdown/1) |> Enum.join("\r")
+          [text("#{k}:"), space(), text(DateTime.to_string(dt))]
+
+        String.starts_with?(v_str, ["http://", "https://"]) ->
+          [text("#{k}:"), space(), link(v_str)]
+
+        v_str =~ "\n" ->
+          [text("#{k}:"), line_break(), text(v_str)]
 
-  def to_markdown(%{} = row) do
-    bullet = "• "
-    indent = "    "
+        true ->
+          [text("#{k}:"), space(), text(v_str)]
+      end
+    end)
+    |> Enum.intersperse([line_break()])
+    |> List.flatten()
+  end
+
+  defp link(v) do
+    %{type: "link", url: v}
+  end
 
-    text =
-      row
-      |> Enum.sort_by(fn {k, _} -> k end)
-      |> Enum.map_join("\r" <> indent, fn {k, v} -> "#{k}: #{v}" end)
+  defp space do
+    %{type: "text", text: " "}
+  end
+
+  defp line_break do
+    %{type: "text", text: "\n"}
+  end
 
-    bullet <> text
+  defp text(v) do
+    %{type: "text", text: v}
   end
+
+  defp stringify(v) when is_integer(v) do
+    Integer.to_string(v)
+  end
+
+  defp stringify(v) when is_float(v) do
+    Float.to_string(v)
+  end
```

Comment:
These two aren't really needed as `Jason.encode!/1` should work on these as well. It doesn't work properly in such case?

Reply:
this is needed for the checks here, as the functions are are strict typed
https://github.com/Logflare/logflare/commit/0f884ec71dae96827f055e494ae30d5d71afb6d1